File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesReviews.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.page.PagedResponse;
import dev.reuss.tmdb.common.review.Review;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Reviews for a TMDB TV series.
 *
 * @param id           TMDB TV series id
 * @param page         current page
 * @param results      TV series reviews
 * @param totalPages   total pages
 * @param totalResults total results
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesReviews(
        int id,

        int page,

        List<Review> results,

        @JsonProperty("total_pages")
        int totalPages,

        @JsonProperty("total_results")
        int totalResults
) implements PagedResponse<Review>, TmdbModel {

    public TvSeriesReviews {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesReviews.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.page.PagedResponse
import dev.reuss.tmdb.common.review.Review

/**
 * Reviews for a TMDB TV series.
 *
 * @property id TMDB TV series id
 * @property page One-based index of this result page.
 * @property results TV series reviews
 * @property totalPages Total number of result pages reported by TMDB.
 * @property totalResults Total number of matching results reported by TMDB.
 */
@JvmRecord
data class TvSeriesReviews(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("page")
    override val page: Int,
    @all:JsonProperty("results")
    override val results: List<Review> = emptyList(),
    @all:JsonProperty("total_pages")
    override val totalPages: Int,
    @all:JsonProperty("total_results")
    override val totalResults: Int,
) : PagedResponse<Review>,
    TmdbModel
